tests/contrib/test_viz.py using mpl_mark_image_compare fail with matplotlib 3.11

### Summary

The tests in `test/contrib/test_viz.py` fail with the image comparison yielding RMS values of up to ~21 using python 3.13.3 and matplotplib 3.11.0, although the images look the same to me.

The solution, as discussed on the pyhf slack, is to add a tolerance to the image comparison,
i.e. use
```python
@pytest.mark.mpl_image_compare(tolerance=25)
def test_plot_results(datadir):
...
```


### OS / Environment

```console
PRETTY_NAME="Ubuntu 22.04.5 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.5 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=jammy
```

### Steps to Reproduce


<!--- ...or if you have a failing CLI command paste it between the quotes below -->
```console
nox --session tests --python 3.13 -- contrib
```


### File Upload (optional)

_No response_

### Expected Results

Expected test to pass.

### Actual Results

```console
nox > Running session tests-3.13
nox > Reusing existing virtual environment at .nox/tests-3-13.
nox > python -m pip install --upgrade --editable '.[all]' --group test
nox > python -m pip install --upgrade pytest
nox > python -m pip install --upgrade matplotlib
nox > pytest tests/contrib --mpl --mpl-baseline-path tests/contrib/baseline --mpl-generate-summary html
================================================================ test session starts ================================================================
platform linux -- Python 3.13.13, pytest-9.1.1, pluggy-1.6.0
benchmark: 5.2.3 (defaults: timer=time.perf_counter disable_gc=False min_rounds=5 min_time=0.000005 max_time=1.0 calibration_precision=10 warmup=False warmup_iterations=100000)
Matplotlib: 3.11.0
Freetype: 2.14.3
rootdir: /home/iwsatlas1/kuhlmann/DEMOS/pyhf
configfile: pyproject.toml
plugins: console-scripts-1.4.1, socket-0.8.0, benchmark-5.2.3, requests-mock-1.12.1, mpl-0.19.0, anyio-4.14.0, mock-3.15.1
collected 14 items                                                                                                                                  

tests/contrib/test_contrib_utils.py .....                                                                                                     [ 35%]
tests/contrib/test_viz.py .FFFFFF..                                                                                                           [100%]A summary of test results can be found at: /tmp/tmphx8aaosl/fig_comparison.html


===================================================================== FAILURES ======================================================================
_________________________________________________________________ test_plot_results _________________________________________________________________
Error: Image files did not match.
  RMS Value: 19.695675956638016
  Expected:  
    /tmp/tmphx8aaosl/test_viz.test_plot_results/baseline.png
  Actual:    
    /tmp/tmphx8aaosl/test_viz.test_plot_results/result.png
  Difference:
    /tmp/tmphx8aaosl/test_viz.test_plot_results/result-failed-diff.png
  Tolerance: 
    2
_____________________________________________________________ test_plot_results_no_axis _____________________________________________________________
Error: Image files did not match.
  RMS Value: 20.31634331161874
  Expected:  
    /tmp/tmphx8aaosl/test_viz.test_plot_results_no_axis/baseline.png
  Actual:    
    /tmp/tmphx8aaosl/test_viz.test_plot_results_no_axis/result.png
  Difference:
    /tmp/tmphx8aaosl/test_viz.test_plot_results_no_axis/result-failed-diff.png
  Tolerance: 
    2
___________________________________________________________ test_plot_results_components ____________________________________________________________
Error: Image files did not match.
  RMS Value: 21.84294327313362
  Expected:  
    /tmp/tmphx8aaosl/test_viz.test_plot_results_components/baseline.png
  Actual:    
    /tmp/tmphx8aaosl/test_viz.test_plot_results_components/result.png
  Difference:
    /tmp/tmphx8aaosl/test_viz.test_plot_results_components/result-failed-diff.png
  Tolerance: 
    2
________________________________________________________ test_plot_results_components_no_clb ________________________________________________________
Error: Image files did not match.
  RMS Value: 20.813341124144387
  Expected:  
    /tmp/tmphx8aaosl/test_viz.test_plot_results_components_no_clb/baseline.png
  Actual:    
    /tmp/tmphx8aaosl/test_viz.test_plot_results_components_no_clb/result.png
  Difference:
    /tmp/tmphx8aaosl/test_viz.test_plot_results_components_no_clb/result-failed-diff.png
  Tolerance: 
    2
_______________________________________________________ test_plot_results_components_no_clsb ________________________________________________________
Error: Image files did not match.
  RMS Value: 20.930291727674405
  Expected:  
    /tmp/tmphx8aaosl/test_viz.test_plot_results_components_no_clsb/baseline.png
  Actual:    
    /tmp/tmphx8aaosl/test_viz.test_plot_results_components_no_clsb/result.png
  Difference:
    /tmp/tmphx8aaosl/test_viz.test_plot_results_components_no_clsb/result-failed-diff.png
  Tolerance: 
    2
________________________________________________________ test_plot_results_components_no_cls ________________________________________________________
Error: Image files did not match.
  RMS Value: 12.582164542849268
  Expected:  
    /tmp/tmphx8aaosl/test_viz.test_plot_results_components_no_cls/baseline.png
  Actual:    
    /tmp/tmphx8aaosl/test_viz.test_plot_results_components_no_cls/result.png
  Difference:
    /tmp/tmphx8aaosl/test_viz.test_plot_results_components_no_cls/result-failed-diff.png
  Tolerance: 
    2
============================================================== short test summary info ==============================================================
FAILED tests/contrib/test_viz.py::test_plot_results - Failed: Error: Image files did not match.
FAILED tests/contrib/test_viz.py::test_plot_results_no_axis - Failed: Error: Image files did not match.
FAILED tests/contrib/test_viz.py::test_plot_results_components - Failed: Error: Image files did not match.
FAILED tests/contrib/test_viz.py::test_plot_results_components_no_clb - Failed: Error: Image files did not match.
FAILED tests/contrib/test_viz.py::test_plot_results_components_no_clsb - Failed: Error: Image files did not match.
FAILED tests/contrib/test_viz.py::test_plot_results_components_no_cls - Failed: Error: Image files did not match.
============================================================ 6 failed, 8 passed in 1.85s ============================================================
nox > Command pytest tests/contrib --mpl --mpl-baseline-path tests/contrib/baseline --mpl-generate-summary html failed with exit code 1
nox > Session tests-3.13 failed.
```

### pyhf Version

```console
HEAD
```

### Code of Conduct

- [x] I agree to follow the Code of Conduct

--- a/docs/contributors.rst
+++ b/docs/contributors.rst
@@ -38,3 +38,4 @@ Contributors include:
 - Matthias Bussonnier
 - Peter Fackeldey
 - Alok Kumar
+- Julian Kuhlmann

--- a/tests/contrib/test_viz.py
+++ b/tests/contrib/test_viz.py
@@ -52,7 +52,7 @@ def test_brazil_band_collection(datadir):
     assert brazil_band_collection.axes == ax
 
 
-@pytest.mark.mpl_image_compare
+@pytest.mark.mpl_image_compare(tolerance=25)
 def test_plot_results(datadir):
     data = json.load(datadir.joinpath("hypotest_results.json").open(encoding="utf-8"))
 
@@ -66,7 +66,7 @@ def test_plot_results(datadir):
     return fig
 
 
-@pytest.mark.mpl_image_compare
+@pytest.mark.mpl_image_compare(tolerance=25)
 def test_plot_results_no_axis(datadir):
     data = json.load(datadir.joinpath("hypotest_results.json").open(encoding="utf-8"))
 
@@ -78,7 +78,7 @@ def test_plot_results_no_axis(datadir):
     return fig
 
 
-@pytest.mark.mpl_image_compare
+@pytest.mark.mpl_image_compare(tolerance=25)
 def test_plot_results_components(datadir):
     data = json.load(
         datadir.joinpath("tail_probs_hypotest_results.json").open(encoding="utf-8")
@@ -92,7 +92,7 @@ def test_plot_results_components(datadir):
     return fig
 
 
-@pytest.mark.mpl_image_compare
+@pytest.mark.mpl_image_compare(tolerance=25)
 def test_plot_results_components_no_clb(datadir):
     data = json.load(
         datadir.joinpath("tail_probs_hypotest_results.json").open(encoding="utf-8")
@@ -114,7 +114,7 @@ def test_plot_results_components_no_clb(datadir):
     return fig
 
 
-@pytest.mark.mpl_image_compare
+@pytest.mark.mpl_image_compare(tolerance=25)
 def test_plot_results_components_no_clsb(datadir):
     data = json.load(
         datadir.joinpath("tail_probs_hypotest_results.json").open(encoding="utf-8")
@@ -136,7 +136,7 @@ def test_plot_results_components_no_clsb(datadir):
     return fig
 
 
-@pytest.mark.mpl_image_compare
+@pytest.mark.mpl_image_compare(tolerance=25)
 def test_plot_results_components_no_cls(datadir):
     data = json.load(
         datadir.joinpath("tail_probs_hypotest_results.json").open(encoding="utf-8")

